domain/src/main/java/org/oppia/domain/oppialogger/analytics/AnalyticsController.kt

```diff
@@ -154,4 +173,79 @@ class AnalyticsController @Inject constructor(
       )
       .build()
   }
+
+  /**
+   * Checks network connectivity of the device.
+   * Saves the [eventLog] to the [eventLogStore] in the absence of it.
+   * Uploads to remote service in the presence of it.
+   */
+  private fun checkNetworkAndManageEventLog(eventLog: EventLog) {
+    when (networkConnectionUtil.getCurrentConnectionStatus()) {
+      NONE -> addEventLog(eventLog)
+      else -> eventLogger.logEvent(eventLog)
+    }
+  }
+
+  /**
+   * Adds an event to the storage.
+   * At first, it checks if the size of the store isn't exceeding [eventLogStorageCacheSize]
+   * If the limit is exceeded then the least recent event is removed from the [eventLogStore]
+   * After this, the [eventLog] is added to the store.
+   * */
+  private fun addEventLog(eventLog: EventLog) {
+    eventLogStore.storeDataAsync(updateInMemoryCache = true) { oppiaEventLogs ->
+      val storeSize = oppiaEventLogs.eventLogList.size
+      if (storeSize + 1 > eventLogStorageCacheSize) {
+        val eventLogRemovalIndex = getLeastRecentEventIndex(oppiaEventLogs)
+        if (eventLogRemovalIndex != null) {
+          return@storeDataAsync oppiaEventLogs.toBuilder()
+            .removeEventLog(eventLogRemovalIndex)
+            .addEventLog(eventLog)
+            .build()
+        } else {
+          exceptionLogger.logException(
```

Comment:
Do you think we should always log exceptions passed to the console logger? If so, maybe you could just have a ``consoleLogger.e()`` call with a new ``NullPointerException`` to simplify this & log to both logcat and the logger.

Reply:
Modified, PTAL.

Reply:
The OppiaLogger will solve this case. So I am resolving this, this will be refactored

Reply:
Actually before I resolve, @Sarthak2601 can you add a todo with #1433 as the bug number?

Reply:
Done.